placesautocomplete/src/main/java/com/seatgeek/placesautocomplete/PlacesApi.java

```diff
@@ -144,16 +144,29 @@ public void setLanguageCode(@Nullable String language) {
      */
     public PlacesAutocompleteResponse autocomplete(final String input, final AutocompleteResultType type) throws IOException {
         final String finalInput = input == null ? "" : input;
+        Uri.Builder uriBuilder;
 
         final AutocompleteResultType finalType = type == null ? DEFAULT_RESULT_TYPE : type;
 
-        Uri.Builder uriBuilder = Uri.parse(PLACES_API_BASE)
-                .buildUpon()
-                .appendPath(PATH_AUTOCOMPLETE)
-                .appendPath(PATH_JSON)
-                .appendQueryParameter(PARAMETER_TYPE, finalType.getQueryParam())
-                .appendQueryParameter(PARAMETER_KEY, googleApiKey)
-                .appendQueryParameter(PARAMETER_INPUT, finalInput);
+        if (finalType.getQueryParam() == "no_type") {
```

Comment:
instead of creating a different builder in the `if/else` block here, you should conditionally add the query param to the uri being built as with the other optional parameters below